### DISCUSSION TITLE
Failing to set up AWS auth for a sink?

### DISCUSSION THREAD

**Discussion by nahkampf** [vectordotdev/vector#14419] [Q&A] [answered]
Hi, first off: I'm a total vector noob so bear with me. My googling has yielded no usable answers to this problem, and it might be something completely obvious but I thought I'd ask here.

I'm trying to set up a simple forwarding of our Monolog logfile to cloudwatch, but can't get the AWS auth to work. Here's vectors log message:

```
ERROR vector::cli: Configuration error. error=data did not match any variant of untagged enum AwsAuthentication for key `sinks.cloudwatch`
```

And here's the config:
```json
{
    "sources": {
        "REDACTED_logs": {
            "type": "file",
            "include": [
                "/logs/REDACTED.log"
            ],
            "read_from": "beginning"
        }
    },
    "sinks": {
        "cloudwatch": {
            "type": "aws_cloudwatch_logs",
            "inputs": [
                "REDACTED_logs"
            ],
            "create_missing_group": true,
            "create_missing_stream": true,
            "group_name": "REDACTED-dev",
            "compression": "none",
            "region": "eu-north-1",
            "stream_name": "{{ host }}",
            "auth": {
                "access_key_id": "REDACTED",
                "assume_role": "arn:aws:iam::REDACTED:user/REDACTED",
                "secret_access_key": "REDACTED"
            },
            "encoding": {
                "codec": "text",
                "timestamp_format": "rfc3339"
            }
        }
    }
}

```

**Comment by nahkampf** [comment]
Doh. Was tipped off about `assume_role`, removed it and got it working.

**Comment by jszwedko** [comment, marked as answer]
Hi @nahkampf !

The docs definitely don't make this clear, but `assum_role` isn't compatible with providing static credentials via `access_key_id` and `secret_access_key`. Though, it certainly could be, would you want to open an issue?.

As a work around, you can provide `assume_role` in the config and then provide the AWS credentials via the environment variables: `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`.

**Reply by nahkampf** [reply]
Thanks! Yup, as it happens I could deal without assuming role, but good to know :)
